src/MICore/CommandFactories/MICommandFactory.cs

```diff
@@ -556,7 +562,7 @@ public virtual IEnumerable<Guid> GetSupportedExceptionCategories()
         /// exceptions in the category. Note that this clear all previous exception breakpoints set in this category.</param>
         /// <param name="exceptionBreakpointState">Indicates when the exception breakpoint should fire</param>
         /// <returns>Task containing the exception breakpoint id's for the various set exceptions</returns>
-        public virtual Task<IEnumerable<ulong>> SetExceptionBreakpoints(Guid exceptionCategory, /*OPTIONAL*/ IEnumerable<string> exceptionNames, ExceptionBreakpointStates exceptionBreakpointState)
+        public virtual async Task<IEnumerable<ulong>> SetExceptionBreakpoints(Guid exceptionCategory, /*OPTIONAL*/ IEnumerable<string> exceptionNames, ExceptionBreakpointStates exceptionBreakpointState)
```

Comment:
Just a heads up. `lldb-mi` does not support catchpoints. You will want to keep this method throwing Not Implemented exception and override this method in [gdb.cs](https://github.com/microsoft/MIEngine/blob/main/src/MICore/CommandFactories/gdb.cs)